fdbcli/fdbcli.actor.cpp

```diff
@@ -2980,10 +2980,24 @@ ACTOR Future<int> cli(CLIOptions opt, LineNoise* plinenoise) {
 				if (tokencmp(tokens[0], "kill")) {
 					getTransaction(db, tr, options, intrans);
 					if (tokens.size() == 1) {
-						Standalone<RangeResultRef> kvs = wait( makeInterruptable( tr->getRange(KeyRangeRef(LiteralStringRef("\xff\xff/worker_interfaces"), LiteralStringRef("\xff\xff\xff")), 1) ) );
+						address_interface.clear();
+						state Standalone<RangeResultRef> kvs = wait( makeInterruptable( tr->getRange(KeyRangeRef(LiteralStringRef("\xff\xff/worker_interfaces"), LiteralStringRef("\xff\xff\xff")), 1) ) );
 						for( auto it : kvs ) {
-							auto ip_port = it.key.endsWith(LiteralStringRef(":tls")) ? it.key.removeSuffix(LiteralStringRef(":tls")) : it.key;
-							address_interface[ip_port] = it.value;
+							try {
+								state Key ip_port_key = it.key.endsWith(LiteralStringRef(":tls")) ? it.key.removeSuffix(LiteralStringRef(":tls")) : it.key;
+								auto parsedAddress = NetworkAddress::parse(ip_port_key.toString());
+								state Value it_value = it.value;
+								choose {
+									// Checking network connection before adding workers in the list (address_interface) of processes that can be killed
+									when (Reference<IConnection> conn = wait( timeout( INetworkConnections::net()->connect(parsedAddress), FLOW_KNOBS->CONNECTION_MONITOR_TIMEOUT, Reference<IConnection>() ) )) {
```

Comment:
no need for a choose-when because there is only one wait statement

Reply:
unfortunately, INetworkConnections::net()->connect() will not allow FlowTransport to use that connection later when we send the kill command. We need to send a message to the ClientWorkerInterface and wait for a reply to make a connection which can be used.

Reply:
The way this is written, we will wait for each connection to be established before starting to establish the next connection. Instead, all futures should be added to a vector of futures, and then use waitForAll to wait for all of the connections simultaneously